src/client.spec.ts

```diff
@@ -142,6 +142,15 @@ describe('ReactSDKClient', () => {
       expect(result.success).toBe(false);
     });
 
+    it('calls fetchqualifiedsegments when a user is set', async () => {
```

Comment:
Should we add some additional tests for the negative cases like 
- when we don't have a userContext or 
- when there's a timeout and we verify that there was a `message` for 'failed to initialize onReady before timeout, either the datafile, user info or qualified segments was not set before the timeout'

Reply:
- we already have a userContext check inside fetchqualified method
- Updated changes for this